dts: Replace deprecated DT_INVENTEK_* macros with DT_ALIAS_*

ulfalizer opened the pull request:
*(This PR is not adding any new aliases. See the commit message and the explanation in a separate comment. It's a standalone improvement, which will make later improvements much easier.)*

scripts/dts/gen_defines.py currently generates prefixes from properties
in aliases/ in two formats:

    DT_ALIAS_<property>_*
    DT_<compatible of pointed-to node>_<property>_*

For example, 'aliases { usart-0 = &usart0; }' generates

    DT_ALIAS_USART_0_*
    DT_NXP_LPC_USART_USART_0_*

when the &usart0 node has 'compatible = "nxp,lpc-usart"'.

The second form exists for backwards compatibility. It's spammy and hard
to read, and easy to confuse for DT_\<compatible>\_\<unit address>_*, which
also gets generated for devicetree nodes.

The code in gen_defines.py that generates the deprecated form has a
'# TODO: See if we can remove or deprecate this form' Comment over it.

Update all macros that come from /aliases to use the DT_ALIAS_* form,
for macros prefixed with DT_INVENTEK_*.

ulfalizer committed:
dts: Replace deprecated DT_INVENTEK_* macros with DT_ALIAS_*

scripts/dts/gen_defines.py currently generates prefixes from properties
in aliases/ in two formats:

    DT_ALIAS_<property>_*
    DT_<compatible of pointed-to node>_<property>_*

For example, 'aliases { usart-0 = &usart0; }' generates

    DT_ALIAS_USART_0_*
    DT_NXP_LPC_USART_USART_0_*

when the &usart0 node has 'compatible = "nxp,lpc-usart"'.

The second form exists for backwards compatibility. It's spammy and hard
to read, and easy to confuse for DT_<compatible>_<unit address>_*, which
also gets generated for devicetree nodes.

The code in gen_defines.py that generates the deprecated form has a
'# TODO: See if we can remove or deprecate this form' Comment over it.

Update all macros that come from /aliases to use the DT_ALIAS_* form,
for macros prefixed with DT_INVENTEK_*.

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>

ulfalizer commented:
To clarify, this is not adding any new alias macros. All of the replaced macros were already being generated from aliases. This is just converting from an old and bad form of macros generated from aliases to the better `DT_ALIAS_*` form.

Further improvements might be possible, like switching some of them over to `DT_INST_*`. I want to get rid of the old and bad form of alias macros first though, because it can be done somewhat mechanically (though it's still a lot of tiring manual tinkering).

loicpoulain commented:
Agree, that we should move to DT_INST, but don't want to block this PR though.
LGTM

ulfalizer commented:
> Agree, that we should move to DT_INST, but don't want to block this PR though.
> LGTM

Yeah, this is just a completely standalone improvement, which will also make later improvements much easier, like switching to `DT_INST_*`.

galak committed:
drivers: wifi: eswifi: Convert to use DT_INST defines

Use DT_INST define as this is the preferred set of defines to utilize
for drivers.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak commented:
Add a commit to convert to DT_INST on top of your change.
